cloud/amazon/ec2_placement.py

```diff
@@ -0,0 +1,157 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_placement
+short_description: create, remove or list placement group(s) in EC2.
+description:
+    - Creates, removes and lists placement groups in EC2. This module has a dependency on python-boto.
+version_added: "2.0"
+options:
+  name:
+    description:
+      - The EC2 placement group name.
+    required: False
+    default: null
+    aliases: []
+  strategy:
+    description:
+      - The placement strategy of the new placement group. Currently, the only acceptable value is cluster
+      required: false
+    default: cluster
+    choices: ['cluster']
+    aliases: []
+  state:
+    description:
+      - Whether the placement group should be present or absent. Use list to show all or search specific placement group.
+    required: false
+    default: present
+    choices: ['present', 'absent', 'list']
+    aliases: []
+  region:
+    description:
+      - region in which the resource exists.
+    required: false
+    default: null
+    aliases: ['aws_region', 'ec2_region']
+
+author: "Slawomir Skowron (@szibis)"
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Basic example of adding placement group
+tasks:
+- name: add placement group
+  ec2_placement:
+      name: "example_group"
+      state: "present"
+      region: "us-east-1"
+
+# Basic example for delete placement group
+tasks:
+- name: remove placement group
+  ec2_placement:
+      name: "example_group"
+      state: "absent"
+      region: "us-east-1"
+
+# Basic example for listing all placement groups
+tasks:
+- name: list all placement group
+  ec2_placement:
+      state: "list"
+      region: "us-east-1"
+
+# Basic example for listing specific placement groups
+tasks:
+- name: list placement group
+  ec2_placement:
+      name: "example_group"
+      state: "list"
+      region: "us-east-1"
+'''
+
+import sys
+
+try:
+    import boto.ec2
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            name = dict(required=False),
+            strategy = dict(default='cluster', choices=['cluster']),
+            state = dict(default='present', choices=['present', 'absent', 'list']),
+        )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO:
+        module.fail_json(msg='boto required for this module')
+
+    name = module.params.get('name')
+    strategy = module.params.get('strategy')
+    state = module.params.get('state')
+
+    if not isinstance(name, basestring) and name:
+       module.fail_json(msg="name argument must be string")
+
```

Comment:
you should handle exceptions thrown by boto and then give appropriate error messages on top.